src/test/java/io/github/fileanalysissuite/adaptersdk/impls/jaxrs/AdapterSDKTest.java

```diff
@@ -0,0 +1,134 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.fileanalysissuite.adaptersdk.impls.jaxrs;
+
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.AdapterDescriptor;
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.FailureDetails;
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.FileDataItem;
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.FileListItem;
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.ItemMetadata;
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.RepositoryItem;
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.RepositoryProperties;
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.RepositorySettingDefinition;
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.RetrieveFileListRequest;
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.RetrieveFileDataRequest;
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.RetrieveFileDataResponse;
+import io.github.fileanalysissuite.adaptersdk.impls.jaxrs.internal.serverstubs.model.RetrieveFileListResponse;
+
+import org.junit.Test;
+import org.junit.BeforeClass;
+import static org.junit.Assert.assertEquals;
+
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.io.IOException;
+import java.time.Instant;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+
+public class AdapterSDKTest extends AdapterSDKContainer
+{
+    static RepositoryProperties repositoryProperties;
+    static ItemMetadata itemMetadata;
+    static List<FailureDetails> failureDetails;
+
+    @BeforeClass
+    public static void setup() throws IOException
+    {
+        final Map<String, String> configurationOptions = new HashMap<>();
+        configurationOptions.put("configurationOptions", "configurationOptionsVal");
+
+        final Map<String, String> repositoryOptions = new HashMap<>();
+        repositoryOptions.put("Path", "Fake path");
+
+        itemMetadata = new ItemMetadata()
+            .itemLocation("Fake path")
+            .name("Fake name")
+            .title("Fake title")
+            .size(9L)
+            .createdTime(Instant.MIN.toString())
+            .accessedTime(Instant.MAX.toString())
+            .modifiedTime(Instant.EPOCH.toString())
+            .version(2)
+            .additionalMetadata(Collections.singletonMap("Fake key", "Fake value"));
+
+        repositoryProperties = new RepositoryProperties().configurationOptions(configurationOptions).repositoryOptions(repositoryOptions);
+
```

Comment:
I couldn't figure out how to test `configurationOptions`  in `FakeRepositoryAdapter`. I have been using [FileSystemAdapter](https://github.com/FileAnalysisSuite/restadapter-filesystem/blob/237cc5c5da52d732ac9d844b95d301a085c9eb08/restadapter-filesystem-core/src/main/java/io/github/fileanalysissuite/restadapters/filesystem/core/FileSystemAdapter.java) to get an idea of the implementation but it wasn't used in there.